cli/src/pcluster/aws/resource_groups.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+
+from pcluster.aws.common import AWSExceptionHandler, Boto3Client, Cache
+
+
+class ResourceGroupsClient(Boto3Client):
+    """Implement Resource Groups Boto3 client."""
+
+    def __init__(self):
+        super().__init__("resource-groups")
+
+    @AWSExceptionHandler.handle_client_exception
+    @Cache.cached
+    def get_capacity_reservation_ids_from_group_resources(self, group):
+        """Return a list of capacity reservation ids."""
+        capacity_reservation_ids = []
+        resources = self._client.list_group_resources(Group=group)["Resources"]
+        for resource in resources:
+            if resource["Identifier"]["ResourceType"] == "AWS::EC2::CapacityReservation":
+                capacity_reservation_ids.append(re.match("(.*)(cr-.*)", resource["Identifier"]["ResourceArn"]).group(2))
```

Comment:
It might also be good to clarify what the matches are using the name syntax `?P<match_name>`
```
>>> m = re.match(r"(?P<first_name>\w+) (?P<last_name>\w+)", "Malcolm Reynolds")
>>> m.group('first_name')
'Malcolm'
>>> m.group('last_name')
'Reynolds'
```